ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -91,6 +110,22 @@ public void start() throws ReplicationException {
     vcrClusterParticipant.addListener(new VcrClusterParticipantListener() {
       @Override
       public void onPartitionAdded(PartitionId partitionId) {
+        if (partitionId.isEqual(cloudConfig.vcrHelixUpdaterParitionId)) {
+          vcrHelixUpdateLock.lock();
+          try {
+            if (isAmbryListenerToUpdateVcrHelixRegistered == false) {
+              // Only register the listener once. Unfortunately, we can't unregister a listener, but it's OK since
```

Comment:
I guess it is possible for `isVcrHelixUpdater` to be true in two nodes. For example, lets say at time "t" node A is the owner of partition 1. Due to some partition reassignment (lets say new node addition), partition 1 get assigned to node B at time "t+1". In this case its possible for Node B to get notification of `addPartition` of partition 1 at "t+2" before Node A gets `removePartition` notification (at t+4). If a  main cluster update happens between "t+2" and "t+4", then two nodes might try to update vcr cluster at the same time.

Reply:
I guess its not too much of a problem if in a rare case two nodes update the helix cluster concurrently. Helix should be able to handle this gracefully. But we should remove the comment from here.

Reply:
The rare case here is dangerous, let me double checking with Helix team. I will add the case you described to the comment.

Reply:
Worst case, we need a distributed lock.

Reply:
Going to use helix distributed lock to solve this problem.